src/core/edgeNetwork/injectSendEdgeNetworkRequest.js

```diff
@@ -74,11 +74,18 @@ export default ({
           .call({ error })
           .then(throwError, throwError);
       })
-      .then(networkResponse => {
+      .then(({ parsedBody, statusCode }) => {
         // Note that networkResponse.parsedBody may be undefined if it was a
         // 204 No Content response. That's fine.
-        const response = createResponse(networkResponse.parsedBody);
+        const response = createResponse(parsedBody);
         cookieTransfer.responseToCookies(response);
+
+        if (statusCode >= 400) {
+          const throwError = () => processWarningsAndErrors(response);
+          return onRequestFailureCallbackAggregator
+            .call({ response })
```

Comment:
So it looks like the onRequestFailure lifecycle hook will get called with an object that contains either `error` or `response`. That feels a little weird, but I get it and don't have a better alternative. It would probably be good to document that in `createLifecycle` where we have a little bit of documentation about each hook.